crates/dekaf/src/connector.rs

```diff
@@ -216,6 +216,16 @@ fn constraint_for_projection(
             r#type: constraint::Type::FieldForbidden.into(),
             reason: "Cannot materialize input data to '_is_deleted' when using CDC deletions mode as it will be generated by Dekaf".to_string(),
         }
+    } else if projection
+        .inference
```

Comment:
I was a little confused by this being called `inference`, but I'm 95% sure that this field is actually the shape of this location within the collection's effective read schema, at least based on this logic:

https://github.com/estuary/flow/blob/81f5877a64ff2cd5caa34af866fe888f816acb5d/crates/validation/src/collection.rs#L393-L602

Reply:
yes this was named inference before we had schema inference as such